Docs/types.md

```diff
@@ -0,0 +1,645 @@
+	// group membership to be removed, so it is recommended to use this flag with caution.
+	TolerateMemberNotFoundErrors bool `json:"tolerateMemberNotFoundErrors"`
+
+	// TolerateMemberOutOfScopeErrors determines the behavior of the LDAP sync job when out-of-scope user entries
+	// are encountered. If 'true', an LDAP query for a user that falls outside of the base DN given for the all
+	// user query will be tolerated and only an error will be logged. If 'false', the LDAP sync job will fail
+	// if a user query would search outside of the base DN specified by the all user query. Misconfigured LDAP
+	// sync jobs with this flag set to 'true' can result in groups missing users, so it is recommended to use
+	// this flag with caution.
+	TolerateMemberOutOfScopeErrors bool `json:"tolerateMemberOutOfScopeErrors"`
+}
+
+// ActiveDirectoryConfig holds the necessary configuration options to define how an LDAP group sync interacts with an LDAP
+// server using the Active Directory schema
+type ActiveDirectoryConfig struct {
+	// AllUsersQuery holds the template for an LDAP query that returns user entries.
+	AllUsersQuery LDAPQuery `json:"usersQuery"`
+
+	// UserNameAttributes defines which attributes on an LDAP user entry will be interpreted as its OpenShift user name.
+	UserNameAttributes []string `json:"userNameAttributes"`
+
+	// GroupMembershipAttributes defines which attributes on an LDAP user entry will be interpreted
+	// as the groups it is a member of
+	GroupMembershipAttributes []string `json:"groupMembershipAttributes"`
+}
+
+// AugmentedActiveDirectoryConfig holds the necessary configuration options to define how an LDAP group sync interacts with an LDAP
+// server using the augmented Active Directory schema
+type AugmentedActiveDirectoryConfig struct {
+	// AllUsersQuery holds the template for an LDAP query that returns user entries.
+	AllUsersQuery LDAPQuery `json:"usersQuery"`
+
+	// UserNameAttributes defines which attributes on an LDAP user entry will be interpreted as its OpenShift user name.
+	UserNameAttributes []string `json:"userNameAttributes"`
+
+	// GroupMembershipAttributes defines which attributes on an LDAP user entry will be interpreted
+	// as the groups it is a member of
+	GroupMembershipAttributes []string `json:"groupMembershipAttributes"`
+
+	// AllGroupsQuery holds the template for an LDAP query that returns group entries.
+	AllGroupsQuery LDAPQuery `json:"groupsQuery"`
+
+	// GroupUIDAttributes defines which attribute on an LDAP group entry will be interpreted as its unique identifier.
+	// (ldapGroupUID)
+	GroupUIDAttribute string `json:"groupUIDAttribute"`
+
+	// GroupNameAttributes defines which attributes on an LDAP group entry will be interpreted as its name to use for
+	// an OpenShift group
+	GroupNameAttributes []string `json:"groupNameAttributes"`
+}
+
+// LDAPQuery holds the options necessary to build an LDAP query
+type LDAPQuery struct {
```

Comment:
Double check that LDAP identity provider doesn't use this somewhere. If not, remove along with other ldap group sync stuff